### PR TITLE
Pre-Release 0.3.1 - WiFi refactor and Portrait Mode improvements

### PR DESCRIPTION
**Pull request by PaulGD03**
# Pre-Release Notes: Version 0.3.1

### Portrait Mode Enhancements
- Significant improvements have been made to the portrait mode, which is the default orientation for the Prometheus Mini.
- A new setting has been added to allow manual screen rotation.

### WiFi Screen Refactor
- The WiFi screen has been completely refactored.
- A QR code has been added, which will eventually direct users to the WebUI.

## Bug Fixes and Code Quality
- Various fixes have been implemented to enhance overall code quality.
- Improvements include better code readability and maintainability.

### CODE REVIEW

File: lib/settings/about_screen.dart [hunk 1]
```diff
@@ -39,6 +39,20 @@ Future<String> executeCommand(String command, List<String> arguments) async {
 }
 
 Future<String> getRaspberryPiModel() async {
```
Comment (ada-phillips) [inline lib/settings/about_screen.dart:41]: Should we rename this to `getDeviceModel` or something?

File: lib/settings/settings_screen.dart [hunk 2]
```diff
@@ -39,37 +47,195 @@ class SettingsScreen extends StatefulWidget {
...
+
+  Future<void> launchDisconnectDialog() async {
+    bool shouldDisconnect = await showDialog(
+      barrierDismissible: false,
+      context: context,
+      builder: (BuildContext context) {
+        return AlertDialog(
+          title: const Text('Disconnect from WiFi'),
+          content: const Text(
+              'Do you want to disconnect from the current WiFi network?\nThis may cause any ongoing print jobs to fail.'),
```
Comment (ada-phillips) [inline lib/settings/settings_screen.dart:138]: Is this true? I wouldn't expect so??

File: lib/settings/settings_screen.dart [hunk 3]
```diff
@@ -39,37 +47,195 @@ class SettingsScreen extends StatefulWidget {
...
+    WidgetsBinding.instance.addPostFrameCallback((_) {
+      if (mounted) {
+        isConnected.value = initialConnectionStatus;
+      }
+    });
+  }
+
+  Future<bool> _checkInitialConnectionStatus() async {
+    try {
+      final result = await Process.run('ping', ['-c', '1', 'google.com']);
```
Comment (ada-phillips) [inline lib/settings/settings_screen.dart:76]: I don't think this is an appropriate check for wifi connection status? Internet status, sure, but I might run my machine on an isolated network.

File: lib/settings/settings_screen.dart [hunk 4]
```diff
@@ -39,37 +47,195 @@ class SettingsScreen extends StatefulWidget {
...
+          await Process.run('sudo', [scriptPath, screenRotation, baseUser]);
+
+      if (result.exitCode != 0) {
+        logger.severe('Failed to restart Orion: ${result.stderr}');
+      } else {
+        logger.info('Orion restarted successfully');
+      }
+
+      // Clean up the temporary file
+      await scriptFile.delete();
```
Comment (ada-phillips) [inline lib/settings/settings_screen.dart:196]: Is there a reason we can't just execute the script from the assets dir?

**Review by ada-phillips — COMMENTED** [review]
Looks good! Had a couple comments/suggestions tho

File: lib/settings/about_screen.dart (hunk 1 above)
Comment (PaulGD03) [reply lib/settings/about_screen.dart]: Agree. Resolved by [6fe2d43](https://github.com/TheContrappostoShop/Orion/pull/18/commits/6fe2d43bcfb362630f12385318261b280d0e01be)

File: lib/settings/settings_screen.dart (hunk 2 above)
Comment (PaulGD03) [reply lib/settings/settings_screen.dart:138]: That warning is under consideration that the user might be connected to a remote machine, so disconnecting could have unforseen consequences.

Comment (ada-phillips) [reply lib/settings/settings_screen.dart:138]: But shouldn't odyssey continue running as normal? I wouldn't expect a UI disconnect to do anything, remote or not.

File: lib/settings/settings_screen.dart [hunk 5]
```diff
@@ -39,37 +47,195 @@ class SettingsScreen extends StatefulWidget {
...
+      logger.severe('Failed to check initial connection status: $e');
+      return false;
+    }
+  }
 
   void _onItemTapped(int index) {
     setState(() {
       _selectedIndex = index;
     });
   }
```
Comment (PaulGD03) [inline lib/settings/settings_screen.dart:88]: Agree. Maybe this is a better method?
```suggestion
  Future<bool> _checkInitialConnectionStatus() async {
    try {
      final interfaces = await NetworkInterface.list();
      return interfaces.isNotEmpty;
    } catch (e) {
      logger.severe('Failed to check initial connection status: $e');
      return false;
    }
  }
```

Comment (ada-phillips) [reply lib/settings/settings_screen.dart:88]: Does that only list connected interfaces?

Comment (PaulGD03) [reply lib/settings/settings_screen.dart:88]: Hmm yeah, maybe this solution is a better fit as a local replacement?
```suggestion
  Future<bool> _checkInitialConnectionStatus() async {
    try {
      // Get the default gateway IP address
      final result = await Process.run('ip', ['route', 'show', 'default']);
      if (result.exitCode != 0) {
        logger.severe('Failed to get default gateway: ${result.stderr}');
        return false;
      }

      // Extract the gateway IP address from the command output
      final gateway = RegExp(r'default via (\S+)')
          .firstMatch(result.stdout)
          ?.group(1);
      if (gateway == null) {
        logger.severe('No default gateway found');
```

File: lib/settings/settings_screen.dart (hunk 2 above)
Comment (ada-phillips) [reply lib/settings/settings_screen.dart:138]: Looks good

File: lib/settings/settings_screen.dart (hunk 5 above)
Comment (ada-phillips) [reply lib/settings/settings_screen.dart:88]: That seems good. Have you tested it on a pi and another device?

**Review by ada-phillips — APPROVED** [review]

File: lib/settings/settings_screen.dart (hunk 4 above)
Comment (PaulGD03) [reply lib/settings/settings_screen.dart:196]: I haven't got that approach to work yet, but will try for the next release.

File: lib/settings/settings_screen.dart (hunk 5 above)
Comment (PaulGD03) [reply lib/settings/settings_screen.dart:88]: Tested on Pi, seems to work fine! On other OSes WiFi is not supported, so any exception should be expected. Adding a check to only execute this check on Linux and simply return false on any other OS.

File: lib/settings/settings_screen.dart (hunk 3 above)
Comment (PaulGD03) [reply lib/settings/settings_screen.dart]: Resolved by [f161435](https://github.com/TheContrappostoShop/Orion/pull/18/commits/f161435bb86c4bbea504e99a66269aff109012bc).

**Pull request merged**